$NetBSD: patch-ab,v 1.6 1999/04/01 09:59:04 agc Exp $

Use -fPIC on NetBSD
On Solaris, fix up the tcl library name, so it's the same as NetBSD
for pkgsrc internal use (PLIST entries)
On Solaris, do not use dots in version names for the library (same
reason as previous)
Use -fPIC on Solaris too, otherwise we overflow a 2048 entry table.
Set LDFLAGS to the correct value
Set TCL_SHARED_BUILD to 1.

--- configure.orig	Tue Nov 25 16:31:42 1997
+++ configure	Thu Sep 17 11:36:29 1998
@@ -3781,7 +3781,7 @@
 if eval "test \"`echo '$ac_cv_header_'$ac_safe`\" = yes"; then
   echo "$ac_t""yes" 1>&6
   
-	    SHLIB_CFLAGS="-fpic"
+	    SHLIB_CFLAGS="-fPIC"
 	    SHLIB_LD="ld -Bshareable -x"
 	    SHLIB_LD_LIBS=""
 	    SHLIB_SUFFIX=".so"
@@ -3909,7 +3909,7 @@
 	TCL_LIB_VERSIONS_OK=nodots
 	;;
     SunOS-5*)
-	SHLIB_CFLAGS="-KPIC"
+	SHLIB_CFLAGS="-fPIC"
 	SHLIB_LD="/usr/ccs/bin/ld -G -z text"
 
 	# Note: need the LIBS below, otherwise Tk won't find Tcl's
@@ -3921,6 +3921,9 @@
 	DL_LIBS="-ldl"
 	LD_FLAGS=""
 	LD_SEARCH_FLAGS='-R ${LIB_RUNTIME_DIR}'
+	TCL_SHARED_LIB_SUFFIX='`echo ${VERSION} | tr -d .`.so.1.0'
+	TCL_UNSHARED_LIB_SUFFIX='`echo ${VERSION} | tr -d .`.a'
+	TCL_LIB_VERSIONS_OK=nodots
 	;;
     ULTRIX-4.*)
 	SHLIB_CFLAGS="-G 0"
@@ -4345,6 +4348,8 @@
     MAKE_LIB="ar cr ${TCL_LIB_FILE} \${OBJS}"
 fi
 
+TCL_SHARED_BUILD=1
+
 # Note:  in the following variable, it's important to use the absolute
 # path name of the Tcl directory rather than "..":  this is because
 # AIX remembers this path and will attempt to use it at run-time to look
@@ -4518,7 +4523,7 @@
 s%@BUILD_DLTEST@%$BUILD_DLTEST%g
 s%@DL_LIBS@%$DL_LIBS%g
 s%@DL_OBJS@%$DL_OBJS%g
-s%@LD_FLAGS@%$LD_FLAGS%g
+s%@LD_FLAGS@%$LDFLAGS%g
 s%@MAKE_LIB@%$MAKE_LIB%g
 s%@MATH_LIBS@%$MATH_LIBS%g
 s%@SHLIB_CFLAGS@%$SHLIB_CFLAGS%g
